### PR TITLE
Add previtem/nextitem/changed() to loop context

### PR DESCRIPTION
**Pull request by ThiefMaster**
This should solve the problems related to not being able to keep state across loop iterations (e.g. #641) once and for all.

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Any better suggestions for the names of these attributes?

My first thought was `prev` and `next`, but that could be confusing since there are already `first` and `last` which are booleans.

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Jinja2 Changelog
 ================
 
+Version 2.9.6
+-------------
+(bugfix release, released on February ???? 2017)
```
Comment (untitaker) [inline CHANGES]: Why put this into a bugfix release though? This is not acceptable according to semver.

**Comment by mitsuhiko** [comment]
Maybe `previtem` and `nextitem`?

**Comment by untitaker** [comment]
Full disclaimer, I'm relatively unfamiliar with Jinja internals.

File: CHANGES (hunk 1 above)
Comment (ThiefMaster) [inline CHANGES]: Considering that it fixes a usecase that was broken in 2.9 it seems somewhat appropriate

Comment (untitaker) [inline CHANGES]: Is that now the solution to #641 or are we following the idea from Armin there?

Comment (ThiefMaster) [inline CHANGES]: Just an idea which I think would provide a nice solution to that problem

**Comment by davidism** [comment]
This should still go in even if we go ahead with the namespace idea too. It's a simpler solution to probably the majority of cases.

**Comment by ThiefMaster** [comment]
Maybe we should move this to 2.10 too... doesn't make much sense to have #684 in 2.10 and this one in 2.9

**Comment by davidism** [comment]
I'm fine with that. Can we just release 2.10 now?

File: docs/templates.rst [hunk 2]
```diff
@@ -680,6 +689,30 @@ a bug where in some circumstances it appeared that assignments would work.
 This is not supported.  See :ref:`assignments` for more information about
 how to deal with this.
 
+If all you want to do is checking whether some value has changed since the
```
Comment (davidism) [inline docs/templates.rst]: "is check" instead of "is checking"

Comment (kblomqvist) [inline docs/templates.rst]: "is to check" instead of "is check"

**Comment by alanhamlett** [comment]
For an immediate solution, I created two custom Jinja2 template functions to get/set globals on `flask.g`. It persists variables across loop iterations:

```python
# Getter for Flask.g request global
def get_global(key):
    return flask.g.get(key)
app.jinja_env.globals.update(
    GET_GLOBAL=get_global,
)


# Setter for Flask.g request global
def set_global(key, val):
    setattr(flask.g, key, val)
    return ''  # return empty string to prevent modifying template
app.jinja_env.globals.update(
    SET_GLOBAL=set_global,
)
```

Use in your template like:

```html
{{ SET_GLOBAL('myvar', 'someval') }}
{{ GET_GLOBAL('myvar') }}
```

**Comment by AlJohri** [comment]
@ThiefMaster can this be released with 2.10? I'm really looking forward to this feature in the stable release. thanks!

**Comment by ThiefMaster** [comment]
Yes, it will be in 2.10

**Pull request merged**